src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1157,76 +1140,72 @@ internal static DeclarationModifiers GetModifier(SyntaxKind kind, SyntaxKind con
-                        }
-                        else if(nextToken.Kind == SyntaxKind.NamespaceKeyword)
-                        {
-                            // Error reported in binding
-                            modTok = ConvertToKeyword(this.EatToken());
-                        }
-                        else if (
-                            nextToken.Kind == SyntaxKind.EnumKeyword ||
-                            nextToken.Kind == SyntaxKind.DelegateKeyword ||
-                            (IsPossibleStartOfTypeDeclaration(nextToken.Kind) && GetModifier(nextToken) != DeclarationModifiers.None))
-                        {
-                            // Misplaced partial
-                            // TODO(https://github.com/dotnet/roslyn/issues/22439):
-                            // We should consider moving this check into binding, but avoid holding on to trees
-                            modTok = AddError(ConvertToKeyword(this.EatToken()), ErrorCode.ERR_PartialMisplaced);
-                        }
-                        else
-                        {
-                            return;
-                        }
+                                flags == ScanPartialFlags.PartialType
+                                    ? IsTypeDeclarationStart(this.CurrentToken.Kind) // post-7.3 requirement.
+                                        ? MessageID.IDS_FeaturePartialTypes
+                                        : MessageID.IDS_FeatureRefPartialModOrdering
+                                    : MessageID.IDS_FeaturePartialMethod);
 
-                        break;
+                            break;
+                        }
 
                     case DeclarationModifiers.Ref:
-                        // 'ref' is only a modifier if used on a ref struct
-                        // it must be either immediately before the 'struct'
-                        // keyword, or immediately before 'partial struct' if
-                        // this is a partial ref struct declaration
                         {
-                            var next = PeekToken(1);
-                            if (next.Kind == SyntaxKind.StructKeyword ||
-                                (next.ContextualKind == SyntaxKind.PartialKeyword &&
-                                 PeekToken(2).Kind == SyntaxKind.StructKeyword))
+                            bool isTypeDecl;
+
+                            var nextToken = PeekToken(1);
+
+                            // In the previous version of the language, we required 'ref' to
+                            // appear immediately before 'struct' or 'partial struct'.
+                            var isRefFollowedByStructOrPartialStruct = nextToken.Kind == SyntaxKind.StructKeyword ||
+                                (nextToken.ContextualKind == SyntaxKind.PartialKeyword && this.PeekToken(2).Kind == SyntaxKind.StructKeyword);
```

Comment:
honestly I think this is actually more clear and simpler, IsRefStruct is not used anywhere else and force garbage values to be assigned.